src/app/utils.ts

```diff
@@ -190,3 +195,19 @@ export function noSpecialCharacters(
   const pattern = /[\\/:*?<>|"]/;
   if (pattern.test(control.value)) return { hasSpecialChars: true };
 }
+
+/**
+ * @param userService - The `UserService` instance that provides the user's language preference observable (`lang$`).
+ * @returns A promise that resolves to the best matching language as a string, which will be either 'en', 'de', or the default 'en'.
+ */
+export function getBestMatchLanguage(userService: UserService): string {
```

Comment:
This function is not working in all cases. Please never ever use subscribe to get a value out of an observable!!! This is leading to runtime errors which are very hard to find and very hard to reproduce

Reply:
removed!